modules/homa-lite/src/lib.rs

```diff
@@ -267,6 +281,24 @@ pub mod module {
 			}
 			current_weight
 		}
+
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			// Update the total amount of Staking balance by acrueing the interest periodically.
+			let interest_rate = Self::staking_interest_rate_per_update();
+			if !interest_rate.is_zero()
+				&& n.checked_rem(&T::StakingUpdateFrequency::get())
+					.map_or(false, |n| n.is_zero())
+			{
+				// Inflate the staking total by the interest rate.
+				// This will only fail when current TotalStakingCurrency is 0. In this case it is OK to fail.
+				let _ = Self::update_total_staking_currency_storage(|current| {
+					Ok(current.saturating_add(interest_rate.mul(current)))
+				});
+				<T as Config>::WeightInfo::on_initialize()
+			} else {
+				0
```

Comment:
this should be one storage read

Reply:
Added a new benchmark function to make return weight more accurate